### ISSUE TITLE
regexp not working at some point in the api

### ISSUE THREAD

**Issue by emahuni** [micromatch/anymatch#48]
I have an API that runs in nodejs, and regexp fails to run when the app has finally booted. I figured something is making `instanceof RegExp` to fail. If I do the following from the cli:

```js
node
> anymatch([/conv/], 'conversations') // => true
```
everything is working as intended, but in the app console: 
```js
> anymatch([/conv/], 'conversations') // => false
```

I managed to isolate the issue to this line: https://github.com/micromatch/anymatch/blob/0aeecc2dbe9199bcaca195fcbb74ee7708dbad6a/index.js#L30
it seems like that is producing false even if it is a regexp. So what I did was install another small lib, ['kind-of'](https://www.npmjs.com/package/kind-of) and do this on that line and everything works fine again.
```js
// from this ...
if (matcher instanceof RegExp) { //...
// to ...
if (kindOf(matcher) === 'regexp') { //...
```
This resolves that issue.

I know this issue is not related to the lib per se, but I have tried to isolate code in the API and it seems like there is something deep in some lib in npm that is causing this issue on the RegExp class. It seems time wasting and futile to try and correct it that way, rather than make this lib detect the regexp another way. Will update if I somehow find out what is causing it, coz that in itself is a huge issue that something is messing with the Regexp prototype and causing this issue. 

However, the reason why I am letting you guys know about this is that this lib may suffer with this bug just because of other libs that mess that prototype, meaning it could probably be common in many apps and people just don't know this as the cause. So instead, this lib should just avoid that pitfall and use another way to detect regexps, or just directly do this in this lib:

```js
// shamelessly steal this util function to this lib 😉 beauty of open-source
function isRegexp(val) {
  if (val instanceof RegExp) return true;
  return typeof val.flags === 'string'
    && typeof val.ignoreCase === 'boolean'
    && typeof val.multiline === 'boolean'
    && typeof val.global === 'boolean';
}

// then use it on that offending line
if (isRegexp(matcher)) { //...
```

**Comment by paulmillr** [comment]
Are you running workers?

**Comment by emahuni** [comment]
No I'm not using workers, why?

**Comment by paulmillr** [comment]
This most probably happens because RegExp in the worker context is different from a main window's RegExp. You've mentioned "when you're running app console it fails". Which app console is that? Is it just nodejs cli?

**Comment by paulmillr** [comment]
Basically if this happens, all other instanceof checks are also unreliable in worker context, and many libraries could stop working properly

**Comment by emahuni** [comment]
Oh ok. This is a Node API based on Strapi. But I think what you're saying tallies well. I am creating an REPL in the API where I am doing these tests, this is probably where that RegExp issue is occuring. Let me test by logging within the api without using the console. Will revert with outcome.

Regardless, isn't implementation of a local isRegexp function, like I suggested above, the best way to correct this? Since there really is an issue with RegExp in some contexts.

It'll cater for when RegExp fails, since that's the last resort before it fails.

**Comment by emahuni** [comment]
> This most probably happens because RegExp in the worker context is different from a main window's RegExp. You've mentioned "when you're running app console it fails". Which app console is that? Is it just nodejs cli?

I'm on MacOS

**Comment by emahuni** [comment]
I just tested Regex independent of anymatch and it works just fine in the console, but not in anymatch. Here is a screenshot:
![image](https://user-images.githubusercontent.com/7290508/186145100-81714d1d-cadf-414d-a48b-b03e388e29eb.png)
The one at the bottom (in utils._.anymatch) is the one I modified with that function. It just works without any issues. Everything else in the readme examples works for the anymatch import except for regexp.

**Comment by emahuni** [comment]
Ok so I think I can reproduce this:

```sh
 yarn create strapi-app my-project --quickstart
cd my-project
yarn add anymatch
yarn strapi console
// then test anymatch(/test/, 'testing') there, it won't work, but /test/ instanceof RegExp will work just fine.
```

**Comment by emahuni** [comment]
That's a pull request for that fix, this is the working version.

**Comment by paulmillr** [comment]
Again, what's the point if you'll still have all other libraries fail with their instanceof Uint8Array (etc) checks?

**Comment by emahuni** [comment]
The point is to make this work in projects where this would probably not work. I wasn't looking at Uint8Arrays, I focused on RegExp. I never said it didn't work on those. Everything else I tested worked except for regexp only. 

This is not specific to that strapi only, it's only that I noticed it there and is a reliable way to reproduce it. If you notice something else, strapi itself is working just fine, instanceof regexp is working as expected, as I demoed above, it's actually something in **anymatch** package's dependency graph that has issues. There is some conflict somewhere. That environment is only just making it reproducible coz that's where I noticed it.

### **In other words, this modification makes it work in worker/repl/other weird environments**, with a little more work if that's the case, without which it would otherwise fail.

**Comment by paulmillr** [comment]
> There is some conflict somewhere

Okay, in this case we need to detect the cause of this conflict.

This fixes anymatch for you -- good. The deep reason why the issue appeared in a first place is required. I maintain many other libraries, which can be affected by it. Maybe the problem is not in the anymatch, but in a shitty way the repl is constructed. 

If you are only using anymatch, that's fine. However, if you're using any other libraries, they can also break.

**Comment by emahuni** [comment]
I have confirmed that this has nothing to do with the way the REPL is constructed, do this:
```js
import repl from 'repl';
import anymatch from 'anymatch';

repl.start('> ').context.a = anymatch;
```
then run the file

in the REPL:
```js
a(/test/, 'testing')
// => false
/test/ instanceof RegExp
// => true
```

same result, this has nothing to do with Strapi
